### DISCUSSION TITLE
Twilio Segment layout grid and container

### DISCUSSION THREAD

**Discussion by marektonik** [twilio-labs/paste#3363] [Help: Figma] [answered]
Hello there! We've noticed that there's a saved grid style in the Paste components featuring two options: a 24px gutter or a 16px gutter, both with 12 columns.

Let's start with the first case.
`Left nav + content` — in this case, we primarily need to compose elements in the space to the right side of the navigation. To address this, we've prepared the following grid including left & right margins (36px) and excluding left navigation. Take a look and let us know if you think it's correct:

<img width="1799" alt="Zrzut ekranu 2023-08-1 o 12 59 34" src="https://github.com/twilio-labs/paste/assets/140410476/77eaff79-b96d-459e-8986-4b146e037605">
__________________________

`Wizard` — in this case, we'll need to include left and right margins of 36px to ensure proper spacing. Additionally, we would like to explore the possibility of having two types of containers in the wizards. The reason behind this is that sometimes the content in the wizards can be quite extensive, requiring a lot of horizontal space, while at other times, we are dealing with simple and small forms that don't take up much real estate.

Therefore, we propose implementing the following set of containers for the wizards:

- regular 8 columns - providing a balanced layout for typical content.
- wide 10 columns - allowing more room for expansive content that demands additional space horizontally.

<img width="1596" alt="Zrzut ekranu 2023-08-1 o 13 29 36" src="https://github.com/twilio-labs/paste/assets/140410476/a19fc9dc-52f7-42af-a8ea-b43f7ecb889a">

By incorporating these two types of containers, we can create more flexible and adaptive wizards to cater to various content scenarios. We welcome your thoughts on this approach and whether it aligns with the project's requirements. Please let us know if you have any further suggestions or considerations. Thank you!

**Comment by loreina** [comment, marked as answer]
Hi @marektonik,

The Grid styles in Figma were made as pre-configured options to match the [Grid component](https://paste.twilio.design/components/grid) in code and there are more examples of layout configurations with the component on our docs site. We also have [spacing and layout guidelines](https://paste.twilio.design/foundations/spacing-and-layout) and recommend following an 8-pixel grid for spacing.

For product-specific layout guidance, especially the content area, I'd recommend syncing with @ceyerkes and @kevinbtu as Paste doesn't have specific content area spacing guidelines yet.
